### ISSUE TITLE
训练存在的问题

### ISSUE THREAD

**Issue by prio158** [ppogg/YOLOv5-Lite#84]
训练的时候 P和R 一直为0，即使训练到300个epoch也是0，同样的数据集在原yolov5上就很好，想请问下是什么原因！
![image](https://user-images.githubusercontent.com/72843017/145762185-46be5162-5dbc-4f6e-9526-370b5adff40d.png)

**Comment by ppogg** [comment]
训练时候的超参数yaml文件发一下

**Comment by prio158** [comment]
就是那个v5Lite-s.yaml



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2021年12月14日 21:10 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.******@***.***> |
| 主题 | Re: [ppogg/YOLOv5-Lite] 训练存在的问题 (Issue #84) |

训练时候的超参数yaml文件发一下

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ppogg** [comment]
好的，是超参数，hyp.scratch.yaml贴一下

**Comment by prio158** [comment]
都默认的，没有改
`
# Hyperparameters for COCO training from scratch
# python train.py --batch 40 --cfg yolov5m.yaml --weights '' --data coco.yaml --img 640 --epochs 300
# See tutorials for hyperparameter evolution https://github.com/ultralytics/yolov5#tutorials


lr0: 0.001  # initial learning rate (SGD=1E-2, Adam=1E-3)
lrf: 0.2  # final OneCycleLR learning rate (lr0 * lrf)
momentum: 0.937  # SGD momentum/Adam beta1
weight_decay: 0.0005  # optimizer weight decay 5e-4
warmup_epochs: 3.0  # warmup epochs (fractions ok)
warmup_momentum: 0.8  # warmup initial momentum
warmup_bias_lr: 0.1  # warmup initial bias lr
box: 0.05  # box loss gain
cls: 0.5  # cls loss gain
cls_pw: 1.0  # cls BCELoss positive_weight
obj: 1.0  # obj loss gain (scale with pixels)
obj_pw: 1.0  # obj BCELoss positive_weight
iou_t: 0.20  # IoU training threshold
anchor_t: 4.0  # anchor-multiple threshold
# anchors: 3  # anchors per output layer (0 to ignore)
fl_gamma: 0.0  # focal loss gamma (efficientDet default gamma=1.5)
hsv_h: 0.015  # image HSV-Hue augmentation (fraction)
hsv_s: 0.7  # image HSV-Saturation augmentation (fraction)
hsv_v: 0.4  # image HSV-Value augmentation (fraction)
degrees: 0.0  # image rotation (+/- deg)
translate: 0.1  # image translation (+/- fraction)
scale: 0.5  # image scale (+/- gain)
shear: 0.0  # image shear (+/- deg)
perspective: 0.0  # image perspective (+/- fraction), range 0-0.001
flipud: 0.0  # image flip up-down (probability)
fliplr: 0.5  # image flip left-right (probability)
mosaic: 0.2  # image mosaic (probability)
mixup: 0.0  # image mixup (probability)

`

**Comment by ppogg** [comment]
试一下学习率调高一点

**Comment by prio158** [comment]
好的，多谢！



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2021年12月15日 13:54 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.******@***.***> |
| 主题 | Re: [ppogg/YOLOv5-Lite] 训练存在的问题 (Issue #84) |

试一下学习率调高一点

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by prio158** [comment]
还想问大佬一个问题，一般来说，int8模型的推理速度要更慢些吗？我实际跑出来的结果好像都是要更慢些，那么量化的意义是什么呢？

**Comment by ppogg** [comment]
量化会更快，无符号8进制比浮点半精度小，但是需要看硬件和平台，最常见的板子如树莓派，用3B跑v5lite，你会发现fp16比int8快，4b则是相反，所以新的芯片在指令集和架构上有适配的int8运算的话，一般都会比浮点运算快，

**Comment by prio158** [comment]
嗯嗯，多谢回答！



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2021年12月15日 17:23 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.******@***.***> |
| 主题 | Re: [ppogg/YOLOv5-Lite] 训练存在的问题 (Issue #84) |

量化会更快，无符号8进制比浮点半精度小，但是需要看硬件和平台，最常见的板子如树莓派，用3B跑v5lite，你会发现fp16比int8快，4b则是相反，所以新的芯片在指令集和架构上有适配的int8运算的话，一般都会比浮点运算快，

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by flyskywhy** [comment]
Fix it in https://github.com/flyskywhy/YOLOv5-Lite/commit/fecb2b598ab113fd0b55e381f851127940363f4f
